### PR TITLE
Vectorize the covariance calculation

### PR DESCRIPTION
**Pull request by kemchenj**
I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Review by mvaligursky — COMMENTED** [review]
nice, but even cleaner would be

```
    vec3 M0 = scale.x * rot[0];
    vec3 M1 = scale.y * rot[1];
    vec3 M2 = scale.z * rot[2];

    covA = vec3(
        dot(M0, M0),
        dot(M0, M1),
        dot(M0, M2)
    );

    covB = vec3(
        dot(M1, M1),
        dot(M1, M2),
        dot(M2, M2)
    );
```

**Comment by slimbuck** [comment]
Hi @kemchenj , did you mean to close this PR? Looks like a useful simplification.

File: src/scene/gsplat/shader-generator-gsplat.js [hunk 1]
```diff
@@ -118,26 +118,20 @@ const splatCoreVS = `
...
         covA = vec3(
-            M0 * M0 + M3 * M3 + M6 * M6,
-            M0 * M1 + M3 * M4 + M6 * M7,
-            M0 * M2 + M3 * M5 + M6 * M8
+            dot(M0, M0),
+            dot(M0, M1),
+            dot(M0, M2)
         );
-
+
```
Comment (mvaligursky) [inline src/scene/gsplat/shader-generator-gsplat.js:130]:
```suggestion

```

**Review by mvaligursky — APPROVED** [review]

**Comment by mvaligursky** [comment]
Hi.. I'm happy to accept a new PR that fixes it :)
Thanks!

**Pull request merged**